runners/core-java/src/main/java/org/apache/beam/runners/core/OutputAndTimeBoundedSplittableProcessElementInvoker.java

```diff
@@ -279,9 +320,14 @@ public PaneInfo pane() {
 
     @Override
     public synchronized void updateWatermark(Instant watermark) {
+      // Updating the watermark without any claimed blocks is allowed.
```

Comment:
<!--new_thread; commit:b04f5ed31c0be1f231dd7c1613ea50d31616b127; resolved:0-->
Why ? Should we at least warn ?

Reply:
Clarified in a comment.